src/core/zowe/core_for_zowe_sdk/profile_manager.py

```diff
@@ -58,260 +71,169 @@ def config_appname(self) -> str:
-        """
         try:
-            # try to get the profilename from defaults
-            try:
-                profilename = profile_jsonc["defaults"][profile_type]
-            except KeyError:
-                warnings.warn("Given profile type has no default profilename")
-            else:
-                return profilename
-
-            # iterate through the profile and check if profile is found
-            for (key, value) in profile_jsonc["profiles"].items():
-                try:
-                    temp_profile_type = value["type"]
-                    if profile_type == temp_profile_type:
-                        return key
-                except KeyError:
-                    warnings.warn(f"Profile {key} has no type attribute")
-
-            # if no profile with matching type found, we raise an exception
-            raise ProfileNotFound(
-                profile_name=profile_type,
-                error_msg=f"No profile with matching profile_type {profile_type} found",
+            cfg_profile, cfg_profile_name = cfg.get_profile(
+                profile_name=profile_name, profile_type=profile_type
             )
-        except ProfileNotFound as exc:
-            raise exc
-
-    def load_profile_properties(self, profile_jsonc: dict, profile_name: str) -> dict:
-        """
-        Load exact profile properties (without prepopulated fields from base profile)
-        from the profile dict and populate fields from the secure credentials storage
-        """
-        try:
-            props = profile_jsonc["profiles"][profile_name]["properties"]
-        except KeyError as exc:
-            raise ProfileNotFound(
-                profile_name=profile_name, error_msg=str(exc)
-            ) from exc
-
-        secure_props = self.load_credentials()
-
-        secure_fields: list = profile_jsonc["profiles"][profile_name].get("secure", [])
-
-        # load properties with key as profile.{profile_name}.properties.{*}
-        for (key, value) in secure_props.items():
-            if re.match("profiles\\." + profile_name + "\\.properties\\.[a-z]+", key):
-                property_name = key.split(".")[3]
-                if property_name in secure_fields:
-                    props[property_name] = value
-                    secure_fields.remove(property_name)
-
-        if len(secure_fields) > 0:
-            raise SecureValuesNotFound(secure_fields)
-
-        return props
-
-    def load_base_profile_properties(self, profile_jsonc: dict) -> dict:
-        """
-        Load base profile
-        """
-        base_props: dict = {}
-
-        try:
-            base_profile_name = self.get_profilename_from_profiletype(
-                profile_jsonc=profile_jsonc, profile_type="base"
+        except ProfileNotFound:
+            warnings.warn(
+                f"Profile not found in file '{cfg.filename}', trying to return base profile instead.",
+                ProfileNotFoundWarning,
```

Comment:
- Is the warning here required if the base profile exists ?
- Is there a way to turn off all warnings ?

Reply:
Regarding if there a way to turn off all warnings, I believe there is 🙂 https://stackoverflow.com/questions/14463277/how-to-disable-python-warnings

Would you prefer for the warning here to be removed in the case that a base profile exists to fall back to?

Reply:
Discussed with @shalearkane and agreed it would be nice to have a `show_warnings` argument on the ProfileManager constructor which defaults to `true`, and if set to `false` will disable all the warnings.